app/modules/code_provider/code_provider_controller.py

```
@@ -142,6 +142,28 @@ async def get_branch_list(self, repo_name: str) -> Dict[str, Any]:
+
+        if provider_type == "github":
+            github_service = GithubService(self.db)
+            return await github_service.get_repo_structure_json(repo_name, branch_name)
+
+        try:
+            provider = CodeProviderFactory.create_provider_with_fallback(repo_name)
+            # Use the provider to get repository structure
+            structure = provider.get_repository_structure(
+                repo_name=repo_name, path="", ref=branch_name, max_depth=4
+            )
+            return structure
```

Ruff (0.14.8)

Consider moving this statement to an `else` block

(TRY300)